app/models/middleware_performance.rb

```diff
@@ -0,0 +1,74 @@
+class MiddlewarePerformance < ActsAsArModel
+  set_columns_hash(
+    :id            => :string,
+    :name          => :string,
+    :resource_type => :string,
+    :feed          => :string,
+    :ems_ref       => :string,
+    :start_time    => :datetime,
+    :end_time      => :datetime
+  )
+
+  alias resource_id id
+  alias resource_name name
+
+  def self.build_results_for_report_middleware(options)
+    start_time, end_time, interval = parse_time_interval(options)
+    results = []
+    model_class.find_each do |ms|
```

Comment:
@josejulio perhaps not literally _return_ but fail with an appropriate message. You could ask on ManageIQ Gitter if that is needed at all and if so, what a good way would be.